internal/app/backend/backend_service_test.go

```diff
@@ -0,0 +1,141 @@
+
+package backend
+
+import (
+	"io"
+	"testing"
+
+	"open-match.dev/open-match/internal/app/frontend"
+
+	"github.com/rs/xid"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"open-match.dev/open-match/internal/app/mmlogic"
+	"open-match.dev/open-match/internal/pb"
+	"open-match.dev/open-match/internal/rpc"
+
+	rpcTesting "open-match.dev/open-match/internal/rpc/testing"
+	statestoreTesting "open-match.dev/open-match/internal/statestore/testing"
+)
+
+func TestAssignTicketsEmptyRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{Assignment: &pb.Assignment{}}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{TicketId: []string{xid.New().String(), xid.New().String()}}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+
+	req := &pb.AssignTicketsRequest{
+		TicketId: []string{xid.New().String(), xid.New().String()},
+		Assignment: &pb.Assignment{
+			Connection: "localhost",
+		},
+	}
+	assignTicketsLoop(tc, req, func(resp *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.NotFound, status.Convert(err).Code())
+	})
+}
+
+func TestAssignTicketsNormalRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	fe := pb.NewFrontendClient(tc.MustGRPC())
+
+	ctResp, err := fe.CreateTicket(tc.Context(), &pb.CreateTicketRequest{Ticket: &pb.Ticket{}})
+	assert.Nil(err)
+
+	req := &pb.AssignTicketsRequest{
+		TicketId: []string{ctResp.Ticket.Id},
+		Assignment: &pb.Assignment{
+			Connection: "localhost",
+		},
+	}
+	assignTicketsLoop(tc, req, func(resp *pb.AssignTicketsResponse, err error) {
+		assert.Equal(&pb.AssignTicketsResponse{}, resp)
+	})
+
+	gtResp, err := fe.GetTicket(tc.Context(), &pb.GetTicketRequest{TicketId: ctResp.Ticket.Id})
+	assert.Nil(err)
+	assert.Equal("localhost", gtResp.Assignment.Connection)
+}
+
+func TestFetchMatchesEmptyRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	fetchMatchesLoop(t, tc, &pb.FetchMatchesRequest{}, func(_ *pb.FetchMatchesResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+}
+
+// TODO: Add FetchMatchesNormalTest when Hostname getter used to initialize mmf service is in
+// https://github.com/GoogleCloudPlatform/open-match/pull/473
+func fetchMatchesLoop(t *testing.T, tc *rpcTesting.TestContext, req *pb.FetchMatchesRequest, handleResponse func(*pb.FetchMatchesResponse, error)) {
+	c := pb.NewBackendClient(tc.MustGRPC())
```

Comment:
need to create this for every time this method is called? Can this be a client you create once and reuse through test?